### DISCUSSION TITLE
Not Get Redirect URL

### DISCUSSION THREAD

**Discussion by Arifcom** [jomweb/billplz#51] [Q&A]
Hi, can you help me?
I have a question, this is my code below:
`$response = $bill->create(
                $id_collection,
                'test@gmail.com',
                null,
                'Buy a product',
                '4000',
                'https://apitest.com/payment/callback',
                'Buy a product',
                [
                    "redirect_url" => "https://test.com/redirect",
                    "reference_1" => "aaa",
                    "reference_2" => "bbb"    
                ]
            );`
Why I can not get redirect_url?
`[email] => test@gmail.com
    [mobile] => 
    [name] => BUY A PRODUCT
    [url] => https://billplz-staging.herokuapp.com/bills/4vv7cfla
    [reference_1_label] => Reference 1
    [reference_1] => aaa
    [reference_2_label] => Reference 2
    [reference_2] => bbb
    [redirect_url] => 
    [callback_url] => https://apitest.com/payment/callbsack
    [description] => Buy a product
    [paid_at] => `
the redirect_url is empty
please help me, thank you...

**Comment by abdusfauzi** [comment]
**Update**: I missed the supplied `redirect_url` value. Sorry.

From the documentation on Billplz.com, you need to supply `redirect_url` because it is an optional parameter.
- https://www.billplz.com/api#v3-bills-create-a-bill

From this documentation in the repo README.md, it mentioned that the `redirect_url` parameter is optional, and can be passed as the 8th parameter.
- https://github.com/jomweb/billplz#create-a-bill

To use the `redirect_url` parameter, you have to make sure the route is available on your system, so that Billplz can redirect to the URL after everything is done on the payment gateway page.

Hope this answers your inquiry.
